File: LinkedList/C/linked_list.c
#include <stdio.h>
#include <stdlib.h>

typedef struct registry {
    int content;
    struct registry *next;
} NODE;

void add(NODE *ll, int value){
    // time complexity: O(n) where n is the linked list height
    if (ll->next == NULL){
        NODE *new_node;
        new_node = malloc(sizeof(NODE));
        new_node->content = value;
        new_node->next = NULL;

        ll->next = new_node;
        return;
    }

    add(ll->next, value);
}

void recursivePrint(NODE *ll){
    if (ll != NULL){
        printf("%d\n", ll->content);
        recursivePrint(ll->next);
    }
};

int main(){

    NODE *ll = NULL;
    ll = malloc(sizeof(NODE));

    ll->content = 0;
    ll->next = NULL;

    add(ll, 1);

    return 0;
};

File: LinkedList/Python/linked_list.py
from node import Node

from print import recursive_print
from add import add

def main():
    linked_list = Node(content=0, next=None)
    
    for i in range(1, 5):
        add(node=linked_list, value=i)
    
    recursive_print(linked_list)

if __name__ == '__main__':
    main()